### PR TITLE
[WIP] DMDc

### PR DESCRIPTION
**Pull request by AlCap23**
Needs tests and maybe callback integration

### CODE REVIEW

File: src/dmdc.jl [hunk 1]
```diff
@@ -0,0 +1,72 @@
+using DifferentialEquations
```
Comment (ChrisRackauckas) [inline src/dmdc.jl]: use OrdinaryDiffEq directly for these

File: examples/DMDc_Examples.jl [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+gr()
+
+# Define measurements from unstable system with known control input
+X = [4 2 1 0.5 0.25; 7 0.7 0.07 0.007 0.0007]
+U = [-4 -2 -1 -0.5]
+B = Float32[1; 0]
+
+# See fail with unknown input
+sys = DMDc(X, U)
+# But with a little more knowledge
```
Comment (ChrisRackauckas) [inline examples/DMDc_Examples.jl]: oh nice, good example!

**Comment by ChrisRackauckas** [comment]
Did you pull that example from the paper? That's a good one to keep in mind as a positive proof that added knowledge can be required for equation discovery.

**Comment by AlCap23** [comment]
From the book of Kutz and Proctor. But it is also in the paper and on their homepage as a Matlab block I think.

**Pull request merged**